aeron-archive/src/main/java/io/aeron/archive/RecordingSession.java

```diff
@@ -260,21 +261,41 @@ else if (image.isEndOfStream() || image.isClosed())
             countedErrorHandler.onError(ex);
             errorMessage = ex.getMessage();
             errorCode = ex.errorCode();
-            state(State.INACTIVE);
+            state(State.INACTIVE, "ArchiveException: " + errorMessage);
         }
         catch (final Exception ex)
         {
             countedErrorHandler.onError(ex);
             errorMessage = ex.getClass().getName() + ": " + ex.getMessage();
-            state(State.INACTIVE);
+            state(State.INACTIVE, errorMessage);
         }
 
         return 1;
     }
 
-    private void state(final State newState)
+    private void state(final State newState, final String reason)
     {
-        //System.out.println("RecordingSession: " + state + " -> " + newState);
+        logStateChange(state, newState, recordingId, null != image ? image.position() : NULL_POSITION, reason);
         state = newState;
     }
+
+    @SuppressWarnings("unused")
+    private void logStateChange(
+        final State oldState,
+        final State newState,
+        final long recordingId,
+        final long position,
+        final String reason)
+    {
+        //System.out.println("RecordingSession: " + state + " -> " + newState);
+    }
+
+    @SuppressWarnings("unused")
+    private void logStarted(
+        final long recordingId,
+        final long controlSessionId,
+        final long correlationId,
+        final String subscriptionChannel)
+    {
+    }
```

Comment:
Drop this method, i.e. this is covered via existing events `CMD_IN_START_RECORDING` and `CMD_IN_START_RECORDING2`.